src/cargo/core/compiler/build_context/target_info.rs

```diff
@@ -375,7 +375,7 @@ impl TargetInfo {
                     suffix,
                     prefix: prefix.clone(),
                     flavor: FileFlavor::DebugInfo,
-                    crate_type: Some(crate_type.clone()),
+                    crate_type: Some(crate_type),
```

Comment:
I somewhat don't like this change, because it means if you need to later use `crate_type`, you have to go up and revert this change.  (Ideally `crate_type` would be `Copy` so it wouldn't matter.)  It's not too important to me, but I do find clippy a little too pedantic at times.